recipes/libyaml/all/test_package/test_package.c

```diff
@@ -0,0 +1,101 @@
+        1, YAML_ANY_MAPPING_STYLE);
+    if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+    yaml_scalar_event_initialize(&event, NULL, (yaml_char_t *)YAML_STR_TAG,
+        (yaml_char_t *)"fruit", strlen("fruit"), 1, 0, YAML_PLAIN_SCALAR_STYLE);
+    if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+    yaml_sequence_start_event_initialize(&event, NULL, (yaml_char_t *)YAML_SEQ_TAG,
+       1, YAML_ANY_SEQUENCE_STYLE);
+    if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+    for (f = data; f->name; f++) {
+        yaml_mapping_start_event_initialize(&event, NULL, (yaml_char_t *)YAML_MAP_TAG,
+            1, YAML_ANY_MAPPING_STYLE);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+        yaml_scalar_event_initialize(&event, NULL, (yaml_char_t *)YAML_STR_TAG,
+            (yaml_char_t *)"name", strlen("name"), 1, 0, YAML_PLAIN_SCALAR_STYLE);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+        yaml_scalar_event_initialize(&event, NULL, (yaml_char_t *)YAML_STR_TAG,
+            (yaml_char_t *)f->name, strlen(f->name), 1, 0, YAML_PLAIN_SCALAR_STYLE);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+        yaml_scalar_event_initialize(&event, NULL, (yaml_char_t *)YAML_STR_TAG,
+            (yaml_char_t *)"color", strlen("color"), 1, 0, YAML_PLAIN_SCALAR_STYLE);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+        yaml_scalar_event_initialize(&event, NULL, (yaml_char_t *)YAML_STR_TAG,
+            (yaml_char_t *)f->color, strlen(f->color), 1, 0, YAML_PLAIN_SCALAR_STYLE);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+        yaml_scalar_event_initialize(&event, NULL, (yaml_char_t *)YAML_STR_TAG,
+            (yaml_char_t *)"count", strlen("count"), 1, 0, YAML_PLAIN_SCALAR_STYLE);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+        sprintf(buffer, "%d", f->count);
+        yaml_scalar_event_initialize(&event, NULL, (yaml_char_t *)YAML_INT_TAG,
+            (yaml_char_t *)buffer, strlen(buffer), 1, 0, YAML_PLAIN_SCALAR_STYLE);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+        yaml_mapping_end_event_initialize(&event);
+        if (!yaml_emitter_emit(&emitter, &event)) goto error;
+    }
+
+    yaml_sequence_end_event_initialize(&event);
+    if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+    yaml_mapping_end_event_initialize(&event);
+    if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+    yaml_document_end_event_initialize(&event, 0);
+    if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+    yaml_stream_end_event_initialize(&event);
+    if (!yaml_emitter_emit(&emitter, &event)) goto error;
+
+    yaml_emitter_delete(&emitter);
+    return 0;
+
+error:
+    fprintf(stderr, "Failed to emit event %d: %s\n", event.type, emitter.problem);
+    yaml_emitter_delete(&emitter);
+    return 1;
+}
+
```

Comment:
Almost unchanged from bincrafters, aside from adding a newline at end of file to please conan center hook.